### PR TITLE
Implements support for per Actor Type configurations of the Actor Runtime

### LINKED ISSUE #383: Add Support for Per Actor Type Configuration
## Describe the proposal
The Dapr Runtime added support for per actor type configurations in 1.7. This support needs to be added to the SDKs.

The new configuration looks like:

```json
{
  "entities":["actorType1", "actorType2"],
  "actorIdleTimeout": "1h",
  "actorScanInterval": "30s",
  "drainOngoingCallTimeout": "30s",
  "drainRebalancedActors": true,
  "reentrancy": {
    "enabled": true,
    "maxStackDepth": 32
  },
  "entitiesConfig": [
      {
          "entities": ["actorType1"],
          "actorIdleTimeout": "1m",
          "drainOngoingCallTimeout": "10s",
          "reentrancy": {
              "enabled": false
          }
      }
  ]
}
```

Implementation can be varied per SDK but should output to the above JSON.

Docs: https://github.com/dapr/docs/pull/2334
Runtime: https://github.com/dapr/dapr/pull/4201

### PR DESCRIPTION
**Pull request by berndverst**
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

# Description

Implements support for per Actor Type configurations of the Actor Runtime

## Issue reference

Fixes #383 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by berndverst** [comment]
Example:

```python
typeConfig1 = ActorTypeConfig(
    'testactor1',
    actor_scan_interval=timedelta(seconds=10),
    reentrancy=ActorReentrancyConfig(enabled=True))
typeConfig2 = ActorTypeConfig(
    'testactor2',
    drain_ongoing_call_timeout=timedelta(seconds=60),
    reminders_storage_partitions=10)

config = ActorRuntimeConfig(
    actor_scan_interval=timedelta(seconds=30),
    actor_type_configs=[typeConfig1, typeConfig2])
```

**Comment by berndverst** [comment]
@halspang please review as the SME for Per Actor Type Configs

@artursouza @mukundansundar please review as approvers :)

File: dapr/actor/runtime/config.py
Comment (mukundansundar) [inline dapr/actor/runtime/config.py:69]:
```
The duration when in the process of draining rebalanced actors. 
```
Please rephrase

File: dapr/actor/runtime/config.py
Comment (mukundansundar) [inline dapr/actor/runtime/config.py:154]: Why can't this be a `Set`? As in specifically avoiding the possibility of duplicate `ActorTypesConfig`s

File: dapr/actor/runtime/config.py
Comment (mukundansundar) [inline dapr/actor/runtime/config.py:196]: Why add a new variable and then convert to list? 
Can't it be 
`configDict['entities'] = list(self._entities)` ?

File: dapr/actor/runtime/config.py
Comment (berndverst) [reply dapr/actor/runtime/config.py:154]: It doesn't need to be one since you always need to specify the entire List anyway.

Also `{ActorTypeConfig('hello', remindersStoragePartitions=1), ActorTypeConfig('hello', remindersStoragePartitions=2)}` would still be a set of 2 items.

We only allow the entire list to be set or update all at once. So there is little risk in inserting duplicates.

File: dapr/actor/runtime/config.py
Comment (berndverst) [reply dapr/actor/runtime/config.py:196]: we need to keep a method local set to dedupe in a method local context, but at the end we must return a list.

File: dapr/actor/runtime/config.py
Comment (berndverst) [reply dapr/actor/runtime/config.py:69]: Copy pasted the message exactly like this. Will update though

**Review by mukundansundar — DISMISSED** [review]
lgtm

**Review by halspang — COMMENTED** [review]
Change looks good to me overall, but do we have any actor e2e tests?

if you have one for resiliency, a good, low effort way to test this is to specify resiliency as disabled in the main config and provide a per type config with it enabled. If the test still works we know dapr is handling the config correctly.

**Comment by berndverst** [comment]
@mukundansundar @artursouza can I get another review? I added an E2E test (which initially failed and after a lot of troubleshooting I found the root cause). This PR is now in excellent shape - needs no further review, but it needs an **approver sign-off**

**Review by artursouza — APPROVED** [review]

**Pull request merged**
